### PR TITLE
cmd/faucet: merge ipfaucet2 branch to develop

### PR DESCRIPTION
**Pull request by qinglin89**
### Description

merge faucet branch into develop branch

### Rationale

N/A

### Example

N/A

### Changes

Notable changes: 
N/A

### CODE REVIEW

**Review by brilliant-lx — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -74,6 +74,8 @@ require (
 	github.com/fatih/structs v1.1.0
 )
 
+require github.com/yusufpapurcu/wmi v1.2.2 // indirect
```
Comment (guagualvcha) [inline go.mod:77]: why move this?

File: docker/Dockerfile [hunk 2]
```diff
@@ -15,7 +15,7 @@ FROM ethereum/solc:0.6.4-alpine as bsc-genesis
 RUN apk add --d --no-cache ca-certificates npm nodejs bash alpine-sdk
 
 RUN git clone https://github.com/binance-chain/bsc-genesis-contract.git /root/genesis \
-    && rm /root/genesis/package-lock.json && cd /root/genesis && npm install
+    && cd /root/genesis && npm install
```
Comment (guagualvcha) [inline docker/Dockerfile:18]: what is the motivation of this?

Comment (BetZzzzz) [reply docker/Dockerfile:18]: this was from rebase https://github.com/bnb-chain/bsc/pull/1361/commits/425843966f85fe9c10ecdd0f3135bbd5710ec9d9

File: go.mod (hunk 1 above)
Comment (qinglin89) [reply go.mod:77]: no special reason, shown in two places after merge, removed one

**Pull request merged**